urdf/urdfdom_compatibility.h.in

```diff
@@ -69,16 +69,25 @@ URDF_TYPEDEF_CLASS_POINTER(Sphere);
 URDF_TYPEDEF_CLASS_POINTER(Visual);
 
 URDF_TYPEDEF_CLASS_POINTER(ModelInterface);
+
+typedef boost::shared_ptr<void> VoidSharedPtr;
```

Comment:
I see. Didn't look that deep into the code. Nevertheless, VoidSharedPtr is a type within ColladaParser and should be declared there and not in the urdf package. You can convert between std::sharedptr and boost::sharedptr like this:

http://stackoverflow.com/questions/12314967/cohabitation-of-boostshared-ptr-and-stdshared-ptr#comment28669000_12315035